### PR TITLE
fixes 4184: no-sniff and check-for-apache-bug for mime sniffing

### PR DESCRIPTION
**Pull request by eleweek**
I tried fixing #4184 , here is the code I have right now. 

I haven't tested it, because I don't know what is the best way to test this part of code. Would like some help with testing this. Should I write an autotest or should I just test manually?

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6171)

### CODE REVIEW

File: components/net/resource_task.rs [hunk 1]
```diff
@@ -99,8 +99,23 @@ pub fn start_sending_sniffed_opt(start_chan: LoadConsumer, mut metadata: Metadat
...
     if opts::get().sniff_mime_types {
         // TODO: should be calculated in the resource loader, from pull requeset #4094
-        let nosniff = false;
-        let check_for_apache_bug = false;
+        let mut nosniff = false;
+        let mut check_for_apache_bug = false;
+
+        if let Some(ref headers) = metadata.headers {
+            if let Some(ref raw_content_type) = headers.get_raw("content-type") {
+                let ref last_raw_content_type = raw_content_type[raw_content_type.len() - 1];
```
Comment (Manishearth) [inline components/net/resource_task.rs]: what if `len()` is zero?

**Comment by Manishearth** [comment]
Review status: all files reviewed, 3 unresolved discussions, all commit checks successful.
Reviewed files:
- components/net/resource_task.rs @ r1

---

<sup>**[components/net/resource_task.rs, line 113 \[r1\]](https://reviewable.io:443/reviews/servo/servo/6171#-Jq5z4vgNBKhycY6X39s)** ([raw file](https://github.com/servo/servo/blob/e840698d4ad4eb1bf3cce688aa99b12dd4fde66d/components/net/resource_task.rs#L113)):</sup>
Shouldn't this be X-content-type-options?

---

<sup>**[components/net/resource_task.rs, line 115 \[r1\]](https://reviewable.io:443/reviews/servo/servo/6171#-Jq5zB4b2SLXEyEjbbg0)** ([raw file](https://github.com/servo/servo/blob/e840698d4ad4eb1bf3cce688aa99b12dd4fde66d/components/net/resource_task.rs#L115)):</sup>
Nit: `nosniff = raw_blah.iter().any(|ref opt| opt == b"nosniff")` is more idiomatic.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6171)

**Comment by Manishearth** [comment]
Filed https://github.com/w3c/web-platform-tests/issues/1851 for the tests

**Comment by jdm** [comment]
In terms of automated tests, there's the `fetch/nosniff` directory in WPT. These can be enabled by adding the dir to http://mxr.mozilla.org/servo/source/tests/wpt/include.ini and run via `./mach test-wpt /fetch/nosniff/`, but we don't enable MIME sniffing by default right now so they'll still fail. You could switch that in http://mxr.mozilla.org/servo/source/components/util/opts.rs#405 of course, but I don't think we actually want to turn it on by default right now.

**Comment by eleweek** [comment]
I updated the PR, I tested the code manually(both flags are set correctly). According to the discussion above there are things that needs to be done before enabling automatic testing, so I didn't do anything about this.

**Comment by Manishearth** [comment]
Review status: :shipit: all files reviewed, all discussions resolved, all commit checks successful.
Reviewed files:
- components/net/resource_task.rs @ r2

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/6171)

**Comment by Manishearth** [comment]
@bors-servo: r+


thanks!

**Pull request merged**